io.py incompatible with PyYAML-6.0

When building meza, this test failure happens:
```
Doctest: meza.io.read_yaml ... FAIL

======================================================================
FAIL: Doctest: meza.io.read_yaml
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/sw/lib/python3.9/doctest.py", line 2205, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for meza.io.read_yaml
  File "/sw/build.build/meza-py39-0.46.0-1/meza-0.46.0/meza/io.py", line 1256, in read_yaml

----------------------------------------------------------------------
File "/sw/build.build/meza-py39-0.46.0-1/meza-0.46.0/meza/io.py", line 1279, in meza.io.read_yaml
Failed example:
    next(records) == {
        'text': 'Chicago Reader',
        'float': 1.0,
        'datetime': dt(1971, 1, 1, 4, 14),
        'boolean': True,
        'time': '04:14:00',
        'date': date(1971, 1, 1),
        'integer': 40}
Exception raised:
    Traceback (most recent call last):
      File "/sw/lib/python3.9/doctest.py", line 1334, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest meza.io.read_yaml[3]>", line 1, in <module>
        next(records) == {
      File "/sw/build.build/meza-py39-0.46.0-1/meza-0.46.0/meza/io.py", line 551, in read_any
        for line in _read_any(f, reader, args, **kwargs):
      File "/sw/build.build/meza-py39-0.46.0-1/meza-0.46.0/meza/io.py", line 470, in _read_any
        for num, line in enumerate(reader(f, *args, **kwargs)):
    TypeError: load() missing 1 required positional argument: 'Loader'
```
In https://github.com/reubano/meza/blob/370c292e1e07e738006c2f57a9ff7399e775df44/meza/io.py#L1289 , changing `yaml.load` to `yaml.safe_load` takes care of the problem (not sure what version yaml.safe_load was introduced).
See yaml/pyyaml#576

--- a/meza/io.py
+++ b/meza/io.py
@@ -1286,7 +1286,7 @@ def read_yaml(filepath, mode="r", **kwargs):
         ...     'integer': 40}
         True
     """
-    return read_any(filepath, yaml.load, mode, **kwargs)
+    return read_any(filepath, yaml.safe_load, mode, **kwargs)
 
 
 def get_text(element):

--- a/requirements.txt
+++ b/requirements.txt
@@ -6,5 +6,5 @@ xlrd>=1.1.0,<2.0.0
 dbfread==2.0.4
 ijson>=2.3,<3.0.0
 beautifulsoup4>=4.6.0,<5.0.0
-PyYAML>=4.2b1,<6.0.0
+PyYAML>=4.2b1,<7.0.0
 pygogo>=0.13.2,<2.0.0

